browser/ui/views/side_panel/brave_side_panel_resize_widget.cc

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/side_panel/brave_side_panel_resize_widget.h"
+
+#include <utility>
+
+#include "brave/browser/ui/views/frame/brave_browser_view.h"
+#include "brave/browser/ui/views/side_panel/brave_side_panel.h"
+#include "ui/views/controls/resize_area.h"
+#include "ui/views/layout/fill_layout.h"
+#include "ui/views/widget/widget.h"
+
+#if defined(USE_AURA)
+#include "ui/aura/window.h"
+#include "ui/views/view_constants_aura.h"
+#endif
+
+SidePanelResizeWidget::SidePanelResizeWidget(
+    BraveSidePanel* panel,
+    BraveBrowserView* browser_view,
+    views::ResizeAreaDelegate* resize_area_delegate)
+    : panel_(panel) {
+  DCHECK(browser_view->GetWidget());
+
+  observations_.AddObservation(panel_);
+  observations_.AddObservation(browser_view->contents_container());
```

Comment:
Out of curiosiy, do we need this when window size changes while side panel stays with the same size?

Reply:
Although side panel's bounds is not changed, its position could be moved when outer sidebar container is only moved.
so resize widget's position should be updated also.